cmd/fits-api/maps.go

```diff
@@ -13,172 +15,173 @@ type st struct {
 	}
 
-	err := map180.ValidBbox(bbox)
+	err = map180.ValidBbox(q.Get("bbox"))
 	if err != nil {
-		return weft.BadRequest(err.Error())
+		return weft.StatusError{Code: http.StatusBadRequest, Err: err}
 	}
 
-	width := 130
+	width, err := valid.ParseWidth(q.Get("width"))
+	if err != nil {
+		return err
+	}
+
+	if width == 0 {
+		width = 130
+	}
 
-	if v.Get("width") != "" {
-		width, err = strconv.Atoi(v.Get("width"))
+	if q.Get("width") != "" {
+		width, err = strconv.Atoi(q.Get("width"))
 		if err != nil {
-			return weft.BadRequest("invalid width.")
+			return weft.StatusError{Code: http.StatusBadRequest, Err: errors.New("invalid width")}
 		}
 	}
 
 	var s []st
 
-	if v.Get("sites") != "" {
-		for _, si := range strings.Split(v.Get("sites"), ",") {
+	if q.Get("sites") != "" {
+		for _, si := range strings.Split(q.Get("sites"), ",") {
 			s = append(s, st{siteID: si})
 		}
 	} else {
-		s = append(s, st{siteID: v.Get("siteID")})
+		s = append(s, st{siteID: q.Get("siteID")})
 	}
 
 	markers := make([]map180.Marker, 0)
 
 	for _, site := range s {
 
-		if res := validSite(site.siteID); !res.Ok {
-			return res
+		err = validSite(site.siteID)
+		if err != nil {
+			return err
 		}
 
 		g, err := geoJSONSite(site.siteID)
 		if err != nil {
-			return weft.ServiceUnavailableError(err)
+			return err
 		}
 
 		m, err := geoJSONToMarkers(g)
 		if err != nil {
-			return weft.ServiceUnavailableError(err)
+			return err
 		}
 		markers = append(markers, m...)
 
 	}
 
-	by, err := wm.SVG(bbox, width, markers, insetBbox)
+	by, err := wm.SVG(q.Get("bbox"), width, markers, q.Get("insetBbox"))
 	if err != nil {
-		return weft.ServiceUnavailableError(err)
+		return err
 	}
 
 	byt := by.Bytes()
 	b.Write(byt)
 
-	return &weft.StatusOK
+	return nil
 }
 
-func siteTypeMap(r *http.Request, h http.Header, b *bytes.Buffer) *weft.Result {
-	if res := weft.CheckQuery(r, []string{}, []string{"typeID", "methodID", "within", "width", "bbox", "insetBbox"}); !res.Ok {
-		return res
+func siteTypeMap(r *http.Request, h http.Header, b *bytes.Buffer) error {
+	q, err := weft.CheckQueryValid(r, []string{"GET"}, []string{}, []string{"typeID", "methodID", "within", "width", "bbox", "insetBbox"}, valid.Query)
+	if err != nil {
+		return err
 	}
-	h.Set("Content-Type", "image/svg+xml")
 
-	v := r.URL.Query()
+	h.Set("Content-Type", "image/svg+xml")
 
-	bbox := v.Get("bbox")
+	err = map180.ValidBbox(q.Get("bbox"))
+	if err != nil {
+		return weft.StatusError{Code: http.StatusBadRequest, Err: err}
+	}
 
-	err := map180.ValidBbox(bbox)
+	width, err := valid.ParseWidth(q.Get("width"))
 	if err != nil {
-		return weft.BadRequest(err.Error())
+		return err
 	}
 
-	var insetBbox, typeID, methodID, within string
-	width := 130
+	if width == 0 {
+		width = 130
+	}
 
-	if v.Get("insetBbox") != "" {
-		insetBbox = v.Get("insetBbox")
+	var typeID, methodID, within string
 
-		err := map180.ValidBbox(insetBbox)
+	if q.Get("insetBbox") != "" {
```

Comment:
map180.ValidBbox deals with empty string. No need to check empty parameter here.